src/key.rs

```diff
@@ -696,13 +696,7 @@ impl KeyPair {
         &mut self.0
     }
 
-    /// Creates a Schnorr [`KeyPair`] directly from generic Secp256k1 secret key.
-    ///
-    /// # Panics
```

Comment:
Why remove the panics documentation?

Reply:
I'm indifferent as to whether we should say "if you use unsafe code to violate the invariants of this library then this specific function will panic". I can see the argument either way.

Reply:
> if you use unsafe code to violate the invariants of this library then this specific function will panic

Exactly, I have never seen this before. I am sure there are other ways in which we can crash this lib using unsafe code. I think we will need this doc in every function API. This function will panic if you provide corrupted secp context.

Reply:
Is this question resolved? I personally was not able to tell why we assert in some places but not others. Is it worth the dev/review time for me to try and work this out - I can put up individual patches so reviewers can check each one, it will require thoughtful reviews though?

Reply:
This is the only doc-comment with a `Panic` heading that I could find. I think we should remove it. The panic documentation is confusing if it can never happen in safe rust code.